### PR TITLE
New init pattern

### PR DESCRIPTION
**Pull request by alliejanoch**
@SamSamskies - Want to take a look? I made you a collaborator so you can merge the pull request.

### CODE REVIEW

**Comment by SamSamskies** [comment]
I don't think this would work because you are exporting a function that doesn't return anything. So when you say...

``` js
var AWS = require('../index')();
```

AWS is being set to undefined.

You could just do what @ssaunders suggested and simply return awsS3Promisified.initialize instead of awsS3Promisified.initialize().

The other suggestion I would make is to pass in a config object rather than 2 strings. Bugs could easily arise by users transposing the arguments.

``` js
initialize: function(config) {
    config = config || {};

    // default expiration time to 8 hours
    this.setExpirationInSeconds(28800);

    if (config.AWS_ACCESS_KEY_ID && config.AWS_SECRET_ACCESS_KEY) {
      AWS.config.update(config);
    }

    return this;
  }
```

Usage:

``` js
var aws = require('aws-s3-promisified')({
  accessKeyId: AWS_ACCESS_KEY_ID,
  secretAccessKey: AWS_SECRET_ACCESS_KEY
});
```

File: index.js [hunk 1]
```diff
@@ -3,231 +3,249 @@ var BluebirdPromise = require('bluebird');
...
-  setExpirationInSeconds: function(t) {
+  function setExpirationInSeconds (t) {
     this.expirationTime = t;
-  },
+  };
 
-  getExpirationInSeconds: function() {
-    return this.expirationTime;
-  },
+  return {
```
Comment (alliejanoch) [inline index.js:16]: @SamSamskies - returning here.

**Comment by alliejanoch** [comment]
There is a return (line 22). I'm pretty sure it does work, all the tests pass.

I'll do the update with the config object.

File: index.js (hunk 1 above)
Comment (SamSamskies) [inline index.js:16]: Ah, I missed it. I guess that works, but it seems a little unconventional. I would separate the return object if you're going to do it like that to make it more modular.

``` js
var awsS3Promisified = {
    setExpirationInSeconds...
    getExpirationInSecond...
    ...
}

module.exports = function (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY) {
    ...
    return awsS3Promisified;
}
```

Comment (SamSamskies) [inline index.js:16]: This way you could easily extract the awsS3Promisified object into a separate file if you want.

**Comment by alliejanoch** [comment]
@SamSamskies - good points on readability/conventionality. I made some updates. I like this way because then it is easy to see the possible operations, right at the top of the file.

**Comment by SamSamskies** [comment]
Yeah, it's easier to understand now. :+1:

**Pull request merged**